src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettings.java

```diff
@@ -237,9 +237,10 @@ public static CustomGeneratorSettings load(World world) {
             settings = fromJson(jsonString);
             if (isOutdated)
                 settings.save(world);
-        } else if (!CustomCubicWorldType.pendingCustomCubicSettingsJsonString.isEmpty()) {
+        } else if (CustomCubicWorldType.createNewWorld && !CustomCubicWorldType.pendingCustomCubicSettingsJsonString.isEmpty()) {
             settings = CustomGeneratorSettings.fromJson(CustomCubicWorldType.pendingCustomCubicSettingsJsonString);
             CustomCubicWorldType.pendingCustomCubicSettingsJsonString = "";
+            CustomCubicWorldType.createNewWorld = false;
```

Comment:
There should be a better way to determine if it's a new world that is used in common event handler.

Reply:
I can suggest to do a same thing but in a `AttachCapabilitiesEvent<World> `handler with high priority. Another option is to retrieve `saveDirName` from `GuiCreateWorld` and create generator settings before World instance creation on `CustomCubicWorldType.onGUICreateWorldPress()`.